DESIGN.md

```diff
@@ -353,6 +353,91 @@ These are done through the MediaWiki API (most likely through a bot account).
 upload these files.
 
+### Editing data
+
+As per [issue
+#32](https://github.com/opentechstrategies/torque/issues/32), users
+can edit data fields (e.g., to fix a typo in a city's name in the
+"City" field) if they have proper permissions.
+
+Because the original proposal data is inhaled from an on-disk
+spreadsheet at startup time and lives in an in-memory data structure,
+edits are persisted "off to the side" as overlays over that original
+data.  The overlay information is then consulted for updates when
+proposal data is loaded.
+
+For simplicity's sake, we store the edits in their own top-level data
+structure (pickled to and from disk just like the original data is).
+This is what that data structure looks like:
+
+```
+edits = {
+    "proposals": {
+        "1": {
+               "Application #":
+                  { "original": ORIGINAL_VALUE,
+                    "edits": [
+                      {
+                        "new_value": "15",
+                        "edit_message": "says reason for the edit",
+                        "edit_timestamp": "2020-07-31",
+                        "editor": "JRandom",
+                        "approver": NULL,
+                        "approval_code": NULL,
+                        "approval_timestamp": NULL,
+                      },
+                      {
+                        "new_value": "9",
+                        "edit_message": "says reason for the edit",
+                        "edit_timestamp": "2020-07-29",
+                        "editor": "JRandom",
+                        "approver": NULL,
+                        "approval_code": NULL,
+                        "approval_timestamp": NULL,
+                      },
+                      ...
+                    ]
+                  }
+                "Name":
+                  { "original": ORIGINAL_VALUE,
+                    "edits": [
+                      {
+                        "new_value": "Fish",
+                        "edit_message": "says reason for the edit",
+                        "edit_timestamp": "2020-08-12",
+                        "editor": "JRandom",
+                        "approver": NULL,
+                        "approval_code": NULL,
+                        "approval_timestamp": NULL,
+                      },
+                      {
+                        "new_value": "Food",
+                        "edit_message": "says reason for the edit",
+                        "edit_timestamp": "2020-08-07",
+                        "editor": "JRandom",
+                        "approver": NULL,
+                        "approval_code": NULL,
+                        "approval_timestamp": NULL,
+                      },
+                      ...
+                    ]
+                  }
+             },
+         "2": { ...},
+         ...}
+      }
+```
+
+The edit subdicts are listed in reverse chronological order by
+`edit_timestamp`.  The `BASE_HASH` is a hash of the original value for
```

Comment:
You say `BASE_HASH` here, but `ORIGINAL_VALUE` up above.  Which way do we want to go?

Reply:
I'd say lets go with `ORIGNAL_VALUE` as that is currently how edits work